tests/Feature/Service/ServiceLiveVouchersPageTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\AdminUser;
+use App\Voucher;
+use App\Http\Requests\VoucherSearchRequest;
+use Tests\StoreTestCase;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+use Illuminate\Support\Facades\Validator;
+
+class ServiceLiveVouchersPageTest extends StoreTestCase
+{
+    use DatabaseMigrations;
+
+    public function testAdminCanViewLiveVouchers()
+    {
+        $this->adminUser = factory(AdminUser::class)->create();
+        $this->actingAs($this->adminUser, 'admin')
+            ->get(route('admin.vouchers.index'))
+            ->assertResponseOk()
+            ->seeInElement('h1', 'View live vouchers')
+            ->seeInElement('button', 'Search')
+            ->seeInElement('a', 'Reset')
+            ;
+    }
+
+    public function testAdminCanSearchASingleLiveVoucher()
+    {
+        $this->rules = (new VoucherSearchRequest())->rules();
```

Comment:
this may be redundant; the test will impicitly test the voucher code.

Reply:
Gone.